### ISSUE TITLE
Using with I2C drivers that take ownership?

### ISSUE THREAD

**Issue by marikka** [eldruin/xca9548a-rs#1]
Thank you for a great crate!

Many drivers for i2c devices take ownership of the i2c instance. Do you have an example of using such drivers? It would seem that switching channels is not possible after the i2c instance is moved into the driver.

My exact use case is taking subsequent measurements from a number of these sensors: https://crates.io/crates/vl53l0x

**Comment by eldruin** [comment]
Glad to hear that you find this crate useful.
You should be able to do that using [shared-bus](https://crates.io/crates/shared-bus).
I will publish an example of doing this soon and will notify you here.

**Comment by marikka** [comment]
Thanks! I was able to use `shared-bus` to make it work. However, I think this is not the best possible abstraction, as one needs to manually switch between channels. Have you considered an alternative API for this crate, where the end-user would supply the channel information during initialization, and the channel switching would be automatically done just before reading/writing?

**Comment by eldruin** [comment]
Yes, I want to add a `split()` method that returns several virtual I2C devices. Then you can use these devices as if there was no switch in between.
This is what I did in the [pcf857x crate](https://crates.io/crates/pcf857x). Here an example of the [`split()`](https://docs.rs/pcf857x/0.2.0/pcf857x/#splitting-device-into-individual-inputoutput-pins-and-setting-them) method.
I was waiting on https://github.com/rust-embedded/embedded-hal/issues/119 to avoid implementing my own Mutex.

**Comment by Kintaro** [comment]
I also tried to use it with shared-bus, but how am I still able to call select_channels on just the handle?

**Comment by eldruin** [comment]
@Kintaro I am not sure I understand the problem. Is your question referred to the `split()` method?

**Comment by eldruin** [comment]
Ok [it took some work](https://github.com/eldruin/xca9548a-rs/compare/366dee530e24b928cfa29aa097bcecff42f8451b..d612d23f50333cbaaf0b2ed3ea0530e72d392a30) but I implemented splitting. I think it turned out quite nice:
```rust
// Talking to two slaves with the same address connected to different ports
// (Whether they share an address or not does not matter)
let i2c_switch = TCA9548A::new(dev, address);
let mut parts = i2c_switch.split();
let slave_address = 0x20;
let data_for_slave = [0xAB, 0xCD];
// Write some data to the slave using normal I2C interface
parts.i2c0.write(slave_address, &data_for_slave).unwrap();
// Read some data from a slave connected to channel 1
let mut read_data = [0; 2];
parts.i2c1.read(slave_address, &mut read_data).unwrap();
```

**Comment by eldruin** [comment]
I have released this in version 0.2.0. Closing.
cc: @marikka @Kintaro
